src/lib.rs

```diff
@@ -78,95 +81,27 @@ impl<R> Volatile<R> {
     /// volatile.write(1);
     /// assert_eq!(volatile.read(), 1);
     /// ```
-    pub const fn new(reference: R) -> Volatile<R> {
+    pub unsafe fn new<A>(pointer: *mut T, access: A) -> Volatile<T, A> {
```

Comment:
Note taking access by value here means that you can't make this method const, is that desirable?

Reply:
Yeah, you should probably go with a design that allows `const fn`. Details of how are not that important, but `const fn` really feels like a must.

Reply:
Agreed, I think I found a good solution for this.